lib/fsfuncs.js

```diff
@@ -0,0 +1,42 @@
+const fs = require('fs').promises;
+
+function mkdirp(path) {
+  return fs.mkdir(path, { recursive: true });
+}
+
+function writeJSON(file, content) {
+  return fs.writeFile(file, JSON.stringify(content));
+}
+
+function readJSON(file) {
+  return fs.readFile(file)
+    .then(string => JSON.parse(string));
+}
+
+function readDirectoryJSON(dir) {
+  return fs.readdir(dir)
+    .then(fileArray => {
+      return fileArray.map(file => readJSON(file));
```

Comment:
This is an array of promises. Use `Promise.all` to wait for all promises in the array to finish

```js
return Promise.all(fileArray.map(file => readJSON(file)));

```